components/cam/src/dynamics/se/dp_coupling.F90

```diff
@@ -113,9 +116,14 @@ subroutine d_p_coupling(phys_state, phys_tend,  pbuf2d, dyn_out)
         ! Map dynamics state to FV physics grid
         !-----------------------------------------------------------------------
         call t_startf('dyn_to_fv_phys')
-        call dyn_to_fv_phys(elem,ps_tmp(1:nphys_sq,:),zs_tmp(1:nphys_sq,:),     &
-                                 T_tmp(1:nphys_sq,:,:),uv_tmp(1:nphys_sq,:,:,:),&
-                                 om_tmp(1:nphys_sq,:,:),q_tmp(1:nphys_sq,:,:,:))
+        if (se_fv_phys_remap_alg == 0) then
+          call dyn_to_fv_phys(elem,ps_tmp(1:nphys_sq,:),zs_tmp(1:nphys_sq,:),     &
+               T_tmp(1:nphys_sq,:,:),uv_tmp(1:nphys_sq,:,:,:),&
+               om_tmp(1:nphys_sq,:,:),q_tmp(1:nphys_sq,:,:,:))
+        else
+          call gfr_dyn_to_fv_phys(par, dom_mt, tl_f, hvcoord, elem, ps_tmp, zs_tmp, &
```

Comment:
Should we try to think of a naming scheme that makes it clearer what the two options are? Maybe something like dyn_to_fv_phys_low-order() and dyn_to_fv_phys_high-order()?
Maybe it's moot since there might not be a good reason to keep both in the long run.

Reply:
There are 5 public API functions that dynamics/se uses. Please feel free to suggest names for each:
```
               gfr_init, &
￼	       gfr_finish, &
￼	       gfr_dyn_to_fv_phys, &
￼	       gfr_fv_phys_to_dyn, &
￼	       gfr_fv_phys_to_dyn_topo,
```
If you would like me to rename the original functions, too, let me know what names you would like.

Reply:
Initially, I thought it makes sense to use some indication of "high order" in the names, whether it's "high_order" or "HO", but that also seems a bit too general if a higher order scheme gets implemented later.  
Would it be accurate to just call this "2nd order" and use "O2" in the front (or back) of the names instead of "gfr"?

Reply:
I'll be sure to mention in the added documentation (but see also [the latex doc](https://acme-climate.atlassian.net/wiki/download/attachments/1010171942/physgrid-ho-remap-doc.pdf?version=4&modificationDate=1565894780686&cacheVersion=1&api=v2)) that the OOA is N for pgN for u,v,T and max(2,N) for Q because of the limiter. So this current method has arbitrarily high order in N. But it's true: other methods might come in later. For example, I'm working on OOA > 1 for pg1 right now, and I might propose a boosted OOA for N < 3 later that would require a new halo exchange pattern to be implemented. On the other hand, I think we can expect any new method to be called through this interface, so I think it's safe to use the names you'd like without concern another interface will be introduced.

Reply:
oh ok, I misunderstood the OOA then. I guess the current name convention is fine then. Maybe it would be better to rename my scheme with something indicating "low order". In any case, let's just keep the current name convention.

Reply:
what is gfr_ for here?

Reply:
(G)LL (F)V (r)emap. I tend to give public functions in a module a common prefix like one does in C; see, e.g., [scalable space-filling curve, sfcmap](https://github.com/E3SM-Project/E3SM/blob/master/components/homme/src/share/spacecurve_mod.F90) and [scalable grid initialization, sgi](https://github.com/E3SM-Project/E3SM/blob/master/components/homme/src/share/scalable_grid_init_mod.F90)